pyteal/ast/widemath.py

```diff
@@ -157,3 +214,45 @@ def has_return(self):
 
 
 WideRatio.__module__ = "pyteal"
+
+
+class WideUint128(MultiValue):
+    @staticmethod
+    def addw(*terms: Expr):
+        if len(terms) < 2:
+            pass
+        return WideUint128(
+            Op.addw, [TealType.uint64, TealType.uint64], args=list(terms)
+        )
+
+    @staticmethod
```

Comment:
@ahangsu For my edification as a new Python programmer:  It seems like you're providing factory methods where annotating with `@staticmethod`.  So that I know how style logic I write, is `@staticmethod` a de facto way to declare factories?

Brief googling pointed me towards `@classmethod` usage:
* https://realpython.com/python-multiple-constructors/#providing-multiple-constructors-with-classmethod-in-python discusses the idea.
* I also found https://stackoverflow.com/questions/136097/difference-between-staticmethod-and-classmethod, which points to https://stackoverflow.com/a/1950927.
* From the reading, it seems to me the primary difference with `@classmethod` is to support factory method construction in subclasses.  Whether it's advantageous to provide subclass factories probably depends on the use case.